src/stores/controller/controllerLogin.ts

```diff
@@ -14,19 +14,22 @@ import { useThemeStore } from '../theme'
 export const useLoginStore = defineStore('controllerLogin', () => {
   const username = ref('')
   const token = ref('')
+  const role = ref('')
 
   const router = useRouter()
 
   const isLoggedIn = computed(() => {
     return !isEmpty(username.value)
   })
+  const isAdmin = computed(() => role.value === 'admin')
 
   const loadUserFromStorage = () => {
     const loginInfo = getJsonFromStorage('controllerLoginInfo')
 
     if (loginInfo) {
       username.value = loginInfo.username
       token.value = loginInfo.token
+      role.value = JSON.parse(atob(loginInfo.token.split('.')[1])).role
```

Comment:
We already have imported `jwt-decode` package to extract token contents. Don't know if it's more reliable than decoding the token manually as you did here, but good to know.